apps/frontend/src/app/5_pages/MarketMakingPage/hooks/useHandleMarketMaking.ts

```diff
@@ -0,0 +1,183 @@
+import { useCallback } from 'react';
+
+import { t } from 'i18next';
+
+import { SupportedTokens, getTokenContract } from '@sovryn/contracts';
+import { Decimal } from '@sovryn/utils';
+
+import { defaultChainId } from '../../../../config/chains';
+
+import {
+  Transaction,
+  TransactionType,
+} from '../../../3_organisms/TransactionStepDialog/TransactionStepDialog.types';
+import { GAS_LIMIT } from '../../../../constants/gasLimits';
+import { getTokenDisplayName } from '../../../../constants/tokens';
+import { useTransactionContext } from '../../../../contexts/TransactionContext';
+import { useAccount } from '../../../../hooks/useAccount';
+import { useGetProtocolContract } from '../../../../hooks/useGetContract';
+import { translations } from '../../../../locales/i18n';
+import { decimalic } from '../../../../utils/math';
+import { prepareApproveTransaction } from '../../../../utils/transactions';
+import { AmmLiquidityPool } from '../utils/AmmLiquidityPool';
+
+export const useHandleMarketMaking = (onComplete: () => void) => {
+  const { account, signer } = useAccount();
+
+  const { setTransactions, setIsOpen, setTitle } = useTransactionContext();
+  const btcWrapperProxyContract = useGetProtocolContract('btcWrapperProxy');
+
+  const onDeposit = useCallback(
+    async (pool: AmmLiquidityPool, amountA: Decimal, amountB: Decimal) => {
+      if (!account || !signer || !btcWrapperProxyContract) {
+        return;
+      }
+
+      const [tokenAContract, tokenBContract] = await Promise.all([
+        getTokenContract(pool.assetA, defaultChainId).then(
+          item => item.address,
+        ),
+        getTokenContract(SupportedTokens.wrbtc, defaultChainId).then(
+          item => item.address,
+        ),
+      ]);
+
+      const minReturn = amountB.sub(amountB.div(100)).toNumber().toFixed(0);
+
+      const transactions: Transaction[] = [];
+
+      if (pool.assetA !== SupportedTokens.rbtc) {
+        const approve = await prepareApproveTransaction({
+          token: pool.assetA,
+          amount: decimalic(amountA.toString()).toString(),
```

Comment:
```suggestion
          amount: amountA.toString()
```
`amountA` is already `Decimal`